cookbooks/aws-parallelcluster-slurm/libraries/update.rb

```diff
@@ -83,3 +84,14 @@ def execute_command(command, user = "root", timeout = 300, raise_on_error = true
 def is_custom_munge_key_updated?
   config_parameter_changed?(%w(DevSettings MungeKeySettings MungeKeySecretArn))
 end
+
+def is_login_nodes_pool_name_updated?
+  config_parameter_changed?(['LoginNodes', 'Pools', 0, 'Name'])
+end
+
+def is_login_nodes_removed?
+  require 'yaml'
+  config = YAML.safe_load(File.read(node['cluster']['cluster_config_path']))
+  previous_config = YAML.safe_load(File.read(node['cluster']['previous_cluster_config_path']))
+  previous_config.dig("LoginNodes") and !config.dig("LoginNodes")
```

Comment:
Does this logic work in case of an `UPDATE_ROLLBACK`? In case of an error in another recipe, and a rollback is triggered, the configs are reversed i.e. `previous_config` becomes `config` and vice versa.

This function will assume login nodes are being added. It's worth testing this manually by mocking an error in another recipe section and observe if the script is removed after the UPDATE_ROLLBACK.

Reply:
I don't see the point.

Case 1:
Create a cluster with LoginNodes, the script will exist. And we update the cluster with removing the LoginNodes section. The script will be deleted. And if the `UPDATE_ROLLBACK`, the LoginNodes will be added back. The other logic will create script back. That's what we want.

Case 2:
Create a cluster without LoginNodes, the script will not exist. We update cluster with adding new LoginNodes section. The script will be created by the other logic. And if `UPDATE_ROLLBACK`, the logic will delete the script. That's what we want too.

Reply:
Case 2.
Thanks for clarifying this. The `previous_config` and `config` will be reversed during rollback but i've just seen the `:delete` action in the update recipe which ensures that the script is removed during a rollback.

Reply:
I guess we can resolve this conversation.